### DISCUSSION TITLE
Clean install Win10 - can not get basic example to work (TypeError: iterable is not iterable)

### DISCUSSION THREAD

**Discussion by BogdanCerovac** [Siteimprove/alfa#724] [Questions] [answered]
Hi, tried to run basic example with Scraper & Frontier but all I get out of it is that iterable is not iterable;

```
TypeError: iterable is not iterable
    at Object.reduce (D:\NodeProj\PoC-CraWWWler\node_modules\@siteimprove\alfa-iterable\src\iterable.js:68:29)
    at fromIterable (D:\NodeProj\PoC-CraWWWler\node_modules\@siteimprove\alfa-list\src\list.js:413:41)
    at Function.from (D:\NodeProj\PoC-CraWWWler\node_modules\@siteimprove\alfa-list\src\list.js:405:16)
    at Function.of (D:\NodeProj\PoC-CraWWWler\node_modules\@siteimprove\alfa-act\src\audit.js:16:50)
    at file:///D:/NodeProj/PoC-CraWWWler/test.js:14:42
    at processTicksAndRejections (node:internal/process/task_queues:94:5)
    at async Function.with (D:\NodeProj\PoC-CraWWWler\node_modules\@siteimprove\alfa-crawler\src\crawler.js:30:20)
```

Tried with versions 0.9.0, 0.10.0 and 0.11.0 - always same result.

Clean installed them multiple times (node LTS and latest as well, with npm and with yarn), same result (Windows 10 64bit pro OS).

Tried even to hack iterable a bit, returning empty array when it failed otherwise but then just got other errors.

Example code copy pasted from site:

```js
import { Audit } from "@siteimprove/alfa-act";
import { Frontier } from "@siteimprove/alfa-frontier";
import { Crawler } from "@siteimprove/alfa-crawler";

import rules from "@siteimprove/alfa-rules";

const frontier = Frontier.of("http://example.com");

Crawler.with(async (crawler) => {
  for await (const result of crawler.crawl(frontier)) {
    for (const input of result) {
      const outcomes = await Audit.of(input, rules).evaluate();
      console.log(outcomes)
    }
  }
});
```

Any ideas?

**Comment by kasperisager** [comment, marked as answer]
Hi there and thanks for reaching out! My best guess as to what is causing this is that this...

```js
import rules from "@siteimprove/alfa-rules";
```

...might be compiling to this...

```js
const rules = require("@siteimprove/alfa-rules");
```

...which will cause errors like what you're seeing. Can you check if that's the case?

**Reply by kasperisager** [reply]
Note that if you're using the native Node.js support for ES modules (through the `.mjs` extension or the `"type": "module"` field) then the two statements in my previous comment will also be equivalent. We compile the packages to CommonJS modules, which adds a compatibility layer for default exports that isn't really compatible with native ES modules. The easiest workaround is probably to use `require()` instead:

```js
const { default: rules } = require("@siteimprove/alfa-rules");
```

If you run your code through a compiler like Babel or TypeScript then the compatibility layer is detected and handled for you. In that case, a plain default import as seen in the code you initially posted will do the trick.

**Reply by BogdanCerovac** [reply]
Aha, OK thank you for this clarification, I thought `"type": "module"` would do the trick alone. Wasn't using transpilers for such a simple example, but looks like I maybe should use them.

It is looking better now, getting generator from outcomes. Thank you again.

**Reply by kasperisager** [reply]
My pleasure, glad you got it sorted! 👏
